FreeRTOS/Test/CMock/event_groups/event_groups_utest.c

```diff
@@ -0,0 +1,608 @@
+/*
```

Comment:
Consider splitting this file up into multiple files so that you can cover the different memory allocation scenarios and share test cases:
configSUPPORT_DYNAMIC_ALLOCATION = 1 && configSUPPORT_STATIC_ALLOCATION = 1
configSUPPORT_DYNAMIC_ALLOCATION = 1 && configSUPPORT_STATIC_ALLOCATION = 0
configSUPPORT_DYNAMIC_ALLOCATION = 0 && configSUPPORT_STATIC_ALLOCATION = 1

I would suggest separate files for xEventGroupCreateStatic  and xEventGroupCreate which also cover vEventGroupDeletethat and verify whether memory has been allocated and if so that the size of the dynamic allocation matches the size of the StaticEventGroup_t type.

You could also explicitly check that the number of calls to malloc and free are equal.

Reply:
The number of calls to malloc and free is equal or not has already checked. I don't think we need run all those combination for all the test cases. We need to test the dynamic create and static create function. All the other functions test cases should exclude the influence of the way of allocation. Since I can have more control over the staic allocation, so I use statically create the event group for all the other funcitons test case .

Reply:
@mingyue86010 : to get 100% line coverage, you will need all three cases. The coverage metrics from lcov are a bit deceiving since they only take into account the specific FreeRTOSConfig.h options used when the target file is built.